package.json

```diff
@@ -35,12 +34,13 @@
     "JC Brand (jcbrand)"
   ],
   "license": "MIT",
-  "main": "dist/strophe.js",
-  "browser": "strophe.js",
+  "main": "dist/strophe.min.js",
   "scripts": {
+    "build": "webpack --mode production",
     "lint": "make jshint",
     "clean": "make clean",
-    "doc": "make doc"
+    "doc": "make doc",
+    "prepare": "yarpm run build"
```

Comment:
It would be maybe more consistent to use the makefile here too. Just call `node_modules/.bin/webpack` inside the makefile. But this is just an idea. I don't know how jcbrand feels about this.

Reply:
Uhm, I could make the "prepare" package.json script build by using the makefile, but that would require the user running it to have all the necessary system depdnencies (gnu make etc) installed; while using a npm script that only uses the package devDependencies, like `webpack` and `yarpm`, guarantees that all the necessary stuff is installed.

Reply:
The `prepare` script is automatically run by npm:
* when publishing, and for this case using the makefile woudln't be much of a problem because only you maintainers can publish and it's assumed that you have all the necessary stuff installed
* when adding strophe as a dependency using a git url (e.g. github:strophe/strophejs#custombranch) and in this case every user could do it, and they are guaranteed to have installed only NPM/YARN and the (dev)dependencies (NPM installs those) and no other external stuff. They could even be using windows. By using NPM-only commands and dependencies, you can define scripts that work for everyone.

I think that's why it's standard to use `package.json` scripts instead of makefiles or other alternatives. Too bad because makefiles are cool and offer nicer features, but package.json scripts are almost universal.

Reply:
I really like GNU Make and using Makefiles, but they have the problem of not being completely platform independent (getting them to work on Windows is a pain) and they introduce more dependencies (as @LeartS mentioned).

So in light of this, it's probably better to use NPM/Yarn scripts. I'm OK with it if we removed the Makefile entirely in favor of them. I consider that out of scope for this PR, but that' my opinion on the matter.